### PR TITLE
New serverless pattern - Dynamic Bedrock model selection with API Gateway and Lambda

### PR DESCRIPTION
**Pull request by sushir-vr**
*Issue #1751, if available:*

*Description of changes:*
This pattern creates an API Gateway REST API with Lambda proxy integration which places invoke calls to Amazon Bedrock models for text and image generation. Bedrock allows the use of various LLMs by providers like AI21 labs, Anthropic etc which can be used to generate texts and images in response to prompts provided by users.

The API generated in this pattern exposes multiple text and image generation models currently available in Bedrock to allow users to send requests to any of the available models.

(New draft for - https://github.com/aws-samples/serverless-patterns/pull/1695)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sushir-vr** [comment]
Created a new PR as per instructions by @boyney123

File: dynamic-bedrock-svls-sam/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Dynamic Bedrock model selection with API Gateway and Lambda
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/README.md:1]:
```suggestion
# Dynamic Amazon Bedrock model selection with API Gateway and Lambda
```

File: dynamic-bedrock-svls-sam/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Dynamic Amazon Bedrock model selection with API Gateway and Lambda
+
+This pattern creates an API Gateway REST API with Lambda proxy integration which places invoke calls to Amazon Bedrock models for text and image generation.
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/README.md:3]:
```suggestion
This pattern creates an API Gateway REST API with Lambda proxy integration which invokes Amazon Bedrock models for text and image generation.
```

File: dynamic-bedrock-svls-sam/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Learn more about this pattern at Serverless Land Patterns: <TBD>
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/README.md:5]:
```suggestion
Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/dynamic-bedrock-svls-sam
```

File: dynamic-bedrock-svls-sam/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
+{
+  "title": "Dynamic Bedrock model selection with API Gateway and Lambda",
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/example-pattern.json:2]:
```suggestion
  "title": "Dynamic Amazon Bedrock model selection with API Gateway and Lambda",
```

File: dynamic-bedrock-svls-sam/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  "description": "Make API calls to generate image and text responses using Amazon Bedrock models with serverless services",
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/example-pattern.json:3]:
```suggestion
  "description": "Generates image or text responses using Amazon Bedrock models with API Gateway and Lambda integrations",
```

File: dynamic-bedrock-svls-sam/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  "language": "Python",
+  "level": "200",
+  "framework": "SAM",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "A REST API is created that has /image and /text resources associated with the available models in Amazon Bedrock for image and text generation respectively. All resources integrate with a Lambda function over proxy integration.",
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/example-pattern.json:10]:
```suggestion
      "A REST API is created that has /image and /text resources associated with the available models in Amazon Bedrock. All resources integrate with a Lambda function over proxy integration.",
```

File: dynamic-bedrock-svls-sam/example-pattern.json [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+          "* Place a request to /image over HTTP POST method with the required prompt as the request body to generate an image using StabilityAI stable diffusion XL model.",
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/example-pattern.json:11]:
```suggestion
          "The /image POST endpoint generates an image using StabilityAI stable diffusion XL model.",
```

File: dynamic-bedrock-svls-sam/example-pattern.json [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+          "* Place a request to /text/model* over HTTP POST method with the required prompt as the request body to generate a text output using various models offered by Anthropic, AI21 labs."
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/example-pattern.json:12]:
```suggestion
          "The /text/model POST method with the required prompt generates a text output using various Bedrock models offered by Anthropic, AI21 labs."
```

File: dynamic-bedrock-svls-sam/example-pattern.json [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/dynamic-bedrock-svls-sam",
+      "templateURL": "serverless-patterns/dynamic-bedrock-svls-sam",
+      "projectFolder": "dynamic-bedrock-svls-sam",
+      "templateFile": "dynamic-bedrock-svls-sam/template.yaml"
```
Comment (boyney123) [inline dynamic-bedrock-svls-sam/example-pattern.json:21]:
```suggestion
      "templateFile": "template.yaml"
```

**Pull request merged**